CSS property update: align-items (#34142)
mdn/content: files/en-us/web/css/align-items/index.md

{{CSSRef}}

> The [CSS](/en-US/docs/Web/CSS) **`align-items`** property sets the {{cssxref("align-self")}} value on all direct children as a group. In flexbox, it controls the alignment of items on the {{glossary("cross axis")}}. In Grid Layout, it controls the alignment of items on the block axis within their {{glossary("grid area")}}.

Reviewer: ```suggestion
The [CSS](/en-US/docs/Web/CSS) **`align-items`** property sets the {{cssxref("align-self")}} value on all direct children as a group. In flexbox, it controls the alignment of items on the {{glossary("cross axis")}}. In Grid Layout, it controls the alignment of items on the block axis within their {{glossary("Grid Areas", "grid area")}}.
```

glossary link here is broken, it looks like it should be "Grid Areas"

{{CSSRef}}

> The [CSS](/en-US/docs/Web/CSS) **`align-items`** property sets the {{cssxref("align-self")}} value on all direct children as a group. In Flexbox, it controls the alignment of items on the {{glossary("cross axis")}}. In Grid Layout, it controls the alignment of items on the block axis within their {{glossary("grid area")}}.

Reviewer: Question after reviewing the other flexbox guide; we sometimes say "flexbox" and other times capitalize it as "Flexbox". Any preference on this in general? Should we be using uppercase "Flexbox" when we touch these docs?

Author: i am going through all the guides one by one, and fixing it to lower case as i go.

Reviewer: OK thank you!

- `self-end`

  - : The items are packed flush to the edge of the alignment container's end side of the item, in the appropriate axis.

- `baseline`, `first baseline`, `last baseline`

  - : All flex items are aligned such that their [flex container baselines](https://drafts.csswg.org/css-flexbox-1/#flex-baselines) align. The item with the largest distance between its cross-start margin edge and its baseline is flushed with the cross-start edge of the line.

- `stretch`

  - : If the items are smaller than the alignment container, auto-sized items will be equally enlarged to fill the container, respecting the items' width and height limits.

- `safe`

  - : Used alongside an alignment keyword. If the chosen keyword means that the item overflows the alignment container causing data loss, the item is instead aligned as if the alignment mode were `start`.

- `unsafe`

  - : Used alongside an alignment keyword. Regardless of the relative sizes of the item and alignment container and whether overflow which causes data loss might happen, the given alignment value is honored.
    There are also two values that are only used with Flexbox:

- `flex-start`

>   - : Used in flex layout only, aligns the flex items flush against the flex container's main-start or cross-start side.

Reviewer: do we have anything to link to from `main-start` and `cross-start` if a reader doesn't know what they are? Same below for `main-end` and `cross-end`.

Author: i linked to the flex model in the sentence above

## Examples

> In this example we have a container with six children. A {{htmlelement("select")}} drop down menu enables toggling the {{cssxref("display")}} of the container between `grid` and `flex`. A second menu enables changing the value of the container's `align-items` property.

Reviewer: I see "dropdown" more often:

```suggestion
In this example we have a container with six children. A {{htmlelement("select")}} dropdown menu enables toggling the {{cssxref("display")}} of the container between `grid` and `flex`. A second menu enables changing the value of the container's `align-items` property.
```

### CSS

We style a the conainer and items in a manner that ensures we have two lines or rows or items. We defined `.flex` and `.grid` classes, which will be applied to the container with JavaScript. They set the {{cssxref("display")}} value of the container, and change its background and border colors providing an additional indicator that the layout has changed. The six flex items each have a different background color, with the 4th item being two lines long and the 6th item having an enlarged font.

```css

> :where(#container) {

Reviewer: do we need `where()`?

Author: the specificity of the `#container` means we can't override any properties with `.flex` and `.grid`

Author: changed to `flex, grid`

  height: 200px;

>   width: 240px;

Reviewer: Want to make it a little wider? 500px container width and 100px wide children?

```suggestion
  width: 500px;
```

Author: good call.

  align-items: initial; /* Change the value in the live sample */
  border: solid 5px transparent;
  gap: 3px;
}

.flex {
  display: flex;
  flex-wrap: wrap;
  background-color: #8c8c9f;
  border-color: magenta;
}

.grid {
  display: grid;

>   grid-template-columns: repeat(auto-fill, 50px);

Reviewer: If you think it makes sense to apply the other width changes, I guess we need it here, also
```suggestion
  grid-template-columns: repeat(auto-fill, 100px);
```

  min-height: 50px;
  font-size: 30px;
}
```

```css hidden
select {
  font-size: 16px;
}

.row {
  margin-top: 10px;
}

div > div {
  box-sizing: border-box;
  border: 2px solid #fff;

>   width: 50px;

Reviewer: See comment above about making it a little wider

```suggestion
  width: 100px;
```

  display: flex;
  align-items: center;
  justify-content: center;
}
```

### HTML

We include a container {{htmlelement("div")}} with six nested `<div>` children. The HTML for the form and the JavaScript that changes the container's class have been hidden for the sake of brevity.

```html
<div id="container" class="flex">
  <div id="item1">1</div>
  <div id="item2">2</div>
  <div id="item3">3</div>

>   <div id="item4">4<br />4</div>

Reviewer: What do you think about this? I skipped the prose when looking at the live example at first and thought it was a typo.

```suggestion
  <div id="item4">4<br />(line break)</div>
```
